### PR TITLE
feat: add native Apple auth to iOS

### PR DESCRIPTION
**Pull request by evgeniy-ac**

### CODE REVIEW

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs [hunk 1]
```diff
@@ -272,4 +322,26 @@ private async Task WriteAutoClosingMessage(CancellationToken cancellationToken)
...
     }
+
+    private async Task AuthenticateSessionWithPrincipal(
+        string sessionId,
+        ClaimsPrincipal principal,
+        CancellationToken cancellationToken)
+    {
+        var oldUser = HttpContext.User;
+        HttpContext.User = principal;
+        try
```
Comment (crui3er) [inline src/dotnet/Users.Service/Controllers/MobileAuthController.cs:333]: Backet should be the same line as try

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs [hunk 2]
```diff
@@ -272,4 +322,26 @@ private async Task WriteAutoClosingMessage(CancellationToken cancellationToken)
...
+        {
+            var helper = Services.GetRequiredService<ServerAuthHelper>();
+            await helper.UpdateAuthState(
+                    new Session(sessionId),
+                    HttpContext,
+                    cancellationToken)
+                .ConfigureAwait(false);
+        }
+        finally
+        {
```
Comment (crui3er) [inline src/dotnet/Users.Service/Controllers/MobileAuthController.cs:343]: Backet should be the same line as finally

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs [hunk 3]
```diff
@@ -153,6 +157,66 @@ public async Task SignIn(string sessionId, string scheme, CancellationToken canc
...
+        var identity = new ClaimsIdentity(options.ClaimsIssuer);
+
+        if (!email.IsNullOrEmpty()) {
+            identity.AddClaim(new Claim(ClaimTypes.Email, email));
+        }
+
+        if (!name.IsNullOrEmpty()) {
+            var names = name.Split(' ');
+            switch (names.Length)
+            {
```
Comment (crui3er) [inline src/dotnet/Users.Service/Controllers/MobileAuthController.cs:199]: Backet should be the same line as switch

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs [hunk 4]
```diff
@@ -153,6 +157,66 @@ public async Task SignIn(string sessionId, string scheme, CancellationToken canc
...
+        using var tokenResponse = await ExchangeCodeAsync(code, options, cancellationToken).ConfigureAwait(false);
+        if (tokenResponse.Error != null) {
+            Logger.LogError(tokenResponse.Error, $"{nameof(SignInAppleWithCode)} error.");
+
+            return BadRequest();
+        }
+
+        var identity = new ClaimsIdentity(options.ClaimsIssuer);
+
+        if (!email.IsNullOrEmpty()) {
```
Comment (crui3er) [inline src/dotnet/Users.Service/Controllers/MobileAuthController.cs:192]: I see that you take email and name from the client. 
For Google I take them from `tokenResponse`. Is it not possible to do it here the same way?

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs (hunk 4 above)
Comment (evgeniy-ac) [reply src/dotnet/Users.Service/Controllers/MobileAuthController.cs:192]: It's the only place where we can get this data
Also it's available only at the very first request

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs (hunk 1 above)
Comment (evgeniy-ac) [reply src/dotnet/Users.Service/Controllers/MobileAuthController.cs:333]: fixes

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs (hunk 2 above)
Comment (evgeniy-ac) [reply src/dotnet/Users.Service/Controllers/MobileAuthController.cs:343]: fixed

File: src/dotnet/Users.Service/Controllers/MobileAuthController.cs (hunk 3 above)
Comment (evgeniy-ac) [reply src/dotnet/Users.Service/Controllers/MobileAuthController.cs:199]: fixed

**Pull request merged**